tests/validate_yaml.py

```diff
@@ -7,27 +7,82 @@
 import yaml
 import sys
 
-def validate_yaml():
+def recursive_merge(d1, d2):
+    '''
+    Recursively merge two dictionaries, which might have lists
+    Not sure if i done this right, but it works
+    '''
+    for k, v in d2.items():
+        if k in d1:
+            if isinstance(v, dict):
+                d1[k] = recursive_merge(d1[k], v)
+            elif isinstance(v, list):
+                d1[k] += v
+            else:
+                d1[k] = v
+        else:
+            d1[k] = v
+    return d1
+
+def validate_jobs(jobs):
+    '''
+    Validate jobs, they must have a kcidb_test_suite mapping
+    '''
+    for name, definition in jobs.items():
+        if definition.get('kind') in ("test", "job"):
+            if not definition.get('kcidb_test_suite'):
+                raise yaml.YAMLError(
+                    f"KCIDB test suite mapping not found for job: {name}'"
+                )
+
+def validate_scheduler_jobs(data):
+    '''
+    Each entry in scheduler have a job, that should be defined in jobs
+    '''
+    sch_entries = data.get('scheduler')
```

Comment:
nit: `s/sch_entries/schedules/`?

btw what happened to code suggestions in the review? I'm no longer able to add ready-to-apply comments :'(

Reply:
done